core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -495,11 +498,22 @@ class DiplomacyManager() {
         if (!hasFlag(DiplomacyFlags.DeclarationOfFriendship))
             revertToZero(DiplomaticModifiers.DeclarationOfFriendship, 1 / 2f) //decreases slowly and will revert to full if it is declared later
 
-        if (otherCiv().isCityState() && otherCiv().cityStateType == CityStateType.Militaristic) {
+        if (otherCiv().isCityState()) {
+            val eraInfo = civInfo.getEraObject()
+
             if (relationshipLevel() < RelationshipLevel.Friend) {
                 if (hasFlag(DiplomacyFlags.ProvideMilitaryUnit)) removeFlag(DiplomacyFlags.ProvideMilitaryUnit)
             } else {
-                if (!hasFlag(DiplomacyFlags.ProvideMilitaryUnit)) setFlag(DiplomacyFlags.ProvideMilitaryUnit, 20)
+                val relevantBonuses = if (relationshipLevel() == RelationshipLevel.Friend) eraInfo.friendBonus[otherCiv().cityStateType.name]
+                                    else eraInfo.allyBonus[otherCiv().cityStateType.name]
+                if (relevantBonuses != null) {
```

Comment:
In my opinion, indentation here goes to deep. Maybe something like
`if (relevantBonuses == null) return`
and later combining the two directly nested `if`-statements would solve that.